apps/frontend/src/app/5_pages/MarketMakingPage/hooks/useGetUserInfo.ts

```diff
@@ -0,0 +1,138 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { ethers } from 'ethers';
+
+import {
+  SupportedTokens,
+  getProtocolContract,
+  getTokenContract,
+} from '@sovryn/contracts';
+import { getProvider } from '@sovryn/ethers-provider';
+import { Decimal } from '@sovryn/utils';
+
+import { defaultChainId } from '../../../../config/chains';
+
+import { useAccount } from '../../../../hooks/useAccount';
+import { useGetProtocolContract } from '../../../../hooks/useGetContract';
+import { asyncCall } from '../../../../store/rxjs/provider-cache';
+import { decimalic } from '../../../../utils/math';
+import { UserInfo } from '../MarketMakingPage.types';
+import { AmmLiquidityPool } from '../utils/AmmLiquidityPool';
+
+export const useGetUserInfo = (pool: AmmLiquidityPool) => {
+  const { account } = useAccount();
+  const { assetA, assetB, poolTokenA, poolTokenB } = pool;
+  const [reward, setReward] = useState('0');
+  const [balanceA, setBalanceA] = useState(Decimal.ZERO);
```

Comment:
```suggestion
  const [balanceA, setBalanceA] = useState<Decimal>(Decimal.ZERO);
```
You need to set the type explicitly so it doesn't show as `any`. Do the same for `balanceB`.